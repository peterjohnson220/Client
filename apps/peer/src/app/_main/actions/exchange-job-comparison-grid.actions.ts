import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

export const LOAD_EXCHANGE_JOB_COMPARISONS  = '[Peer Main/Exchange Job Comparisons] Load Exchange Job Comparisons';
export const LOAD_EXCHANGE_JOB_COMPARISONS_SUCCESS  = '[Peer Main/Exchange Job Comparisons] Load Exchange Job Comparisons Success';
export const LOAD_EXCHANGE_JOB_COMPARISONS_ERROR = '[Peer Main/Exchange Job Comparisons] Load Exchange Job Comparisons Error';

export class LoadExchangeJobComparisons implements Action {
  readonly type = LOAD_EXCHANGE_JOB_COMPARISONS;
}

export class LoadExchangeJobComparisonsSuccess implements Action {
  readonly type = LOAD_EXCHANGE_JOB_COMPARISONS_SUCCESS;

  constructor(public payload: GridDataResult) {}
}

export class LoadExchangeJobComparisonsError implements Action {
  readonly type = LOAD_EXCHANGE_JOB_COMPARISONS_ERROR;
}

export type Actions
  = LoadExchangeJobComparisons
  | LoadExchangeJobComparisonsSuccess
  | LoadExchangeJobComparisonsError;
