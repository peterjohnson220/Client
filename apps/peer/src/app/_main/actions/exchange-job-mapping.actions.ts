import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

export const LOADING_EXCHANGE_JOB_MAPPINGS  = '[Peer Main/Exchange Job Mapping] Loading Exchange Job Mappings';
export const LOADING_EXCHANGE_JOB_MAPPINGS_SUCCESS  = '[Peer Main/Exchange Job Mapping] Loading Exchange Job Mappings Success';
export const LOADING_EXCHANGE_JOB_MAPPINGS_ERROR = '[Peer Main/Exchange Job Mapping] Loading Exchange Job Mappings Error';
export const UPDATE_EXCHANGE_JOB_MAPPINGS_QUERY = '[Peer Main/Exchange Job Mapping] Update Exchange Job Mappings Query';

export class LoadingExchangeJobMappings implements Action {
  readonly type = LOADING_EXCHANGE_JOB_MAPPINGS;

  constructor(public payload: any) {}
}

export class LoadingExchangeJobMappingsSuccess implements Action {
  readonly type = LOADING_EXCHANGE_JOB_MAPPINGS_SUCCESS;

  constructor(public payload: GridDataResult) {}
}

export class LoadingExchangeJobMappingsError implements Action {
  readonly type = LOADING_EXCHANGE_JOB_MAPPINGS_ERROR;
}

export class UpdateExchangeJobMappingsQuery implements Action {
  readonly type = UPDATE_EXCHANGE_JOB_MAPPINGS_QUERY;

  constructor(public payload: string) {}
}

export type Actions
  = LoadingExchangeJobMappings
  | LoadingExchangeJobMappingsSuccess
  | LoadingExchangeJobMappingsError
  | UpdateExchangeJobMappingsQuery;
