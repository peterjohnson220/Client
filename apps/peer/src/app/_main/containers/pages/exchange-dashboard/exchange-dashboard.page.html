<pf-page-with-title-bar [returnUrl]="'/exchanges'">

  <ng-container page-title>
    {{ (exchange$ | async).ExchangeName }} - Dashboard
  </ng-container>

  <ng-container page-actions>
    <button type="button" class="btn btn-primary" (click)="manageJobsClick()">Manage Jobs</button>
  </ng-container>

  <ng-container page-content>
    <div class="flex-column-container">
      <div class="card w-100 h-100">
        <div class="card-header">
          Industries
        </div>
        <div class="card-body">
          <kendo-chart>
            <kendo-chart-legend position="bottom"></kendo-chart-legend>
            <kendo-chart-area background="none"></kendo-chart-area>
            <kendo-chart-tooltip>
              <ng-template kendoChartSeriesTooltipTemplate let-value="value" let-category="category">
                {{ category }} - {{ value | percent: '1.0-2' }}
              </ng-template>
            </kendo-chart-tooltip>
            <kendo-chart-series>
              <kendo-chart-series-item type="donut"
                                       [data]="(industryChartItems$ | async)"
                                       field="Value"
                                       categoryField="Category">
              </kendo-chart-series-item>
            </kendo-chart-series>
          </kendo-chart>
        </div>
      </div>
    </div>
  </ng-container>

</pf-page-with-title-bar>
