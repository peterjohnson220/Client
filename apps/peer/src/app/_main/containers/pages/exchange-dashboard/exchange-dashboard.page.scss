.exchange-dashboard-container {
  height: 100%;
}

.exchange-dashboard-sidebar {
  min-width: 300px;
  max-width: 300px;
}

.exchange-dashboard-chart-container > * {
  flex: 1;
}
