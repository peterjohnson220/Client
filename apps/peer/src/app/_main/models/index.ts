export * from './existing-company.model';
export * from './payfactors-job.model';
