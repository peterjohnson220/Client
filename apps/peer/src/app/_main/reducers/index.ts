import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromExchangeDashboardReducer from './exchange-dashboard.reducer';
import * as fromGridReducer from 'libs/core/reducers/grid.reducer';
import * as fromExchangeReducer from 'libs/features/peer/reducers/exchange.reducer';
import { IFeatureGridState } from 'libs/core/reducers/grid.reducer';

import * as fromExchangeJobMappingGridReducer from './exchange-job-mapping-grid.reducer';
import * as fromExchangeJobMappingInfoReducer from './exchange-job-mapping-info.reducer';
import * as fromPeerParticipantsReducer from './peer-participants.reducer';
import * as fromPfCompaniesExchangeRequestReducer from './exchange-request/pf-companies.reducer';
import * as fromAccessExchangeRequestReducer from './exchange-request/access-exchange-request.reducer';
import * as fromExchangeRequestReducer from './exchange-request.reducer';

// Feature area state
export interface PeerMainState {
  exchange: fromExchangeReducer.State;
  exchangeDashboard: fromExchangeDashboardReducer.State;
  exchangeJobMapping: IFeatureGridState<fromExchangeJobMappingGridReducer.State>;
  exchangeJobMappingInfo: fromExchangeJobMappingInfoReducer.State;
  peerParticipants: fromPeerParticipantsReducer.State;
  accessExchangeRequest: fromAccessExchangeRequestReducer.State;
  pfCompaniesExchangeRequest: fromPfCompaniesExchangeRequestReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  peerMain: PeerMainState;
}

// Feature area reducers
export const reducers = {
  exchange: fromExchangeReducer.reducer,
  exchangeDashboard: fromExchangeDashboardReducer.reducer,
  exchangeJobMapping: fromExchangeJobMappingGridReducer.reducer,
  exchangeJobMappingInfo: fromExchangeJobMappingInfoReducer.reducer,
  peerParticipants: fromPeerParticipantsReducer.reducer,
  accessExchangeRequest: fromAccessExchangeRequestReducer.reducer,
  pfCompaniesExchangeRequest: fromPfCompaniesExchangeRequestReducer.reducer
};

// Select Feature Area
export const selectFeatureAreaState = createFeatureSelector<PeerMainState>('peerMain');

// Feature Selectors
export const selectExchangeState = createSelector(
  selectFeatureAreaState,
  (state: PeerMainState) => state.exchange
);

export const selectExchangeDashboardState = createSelector(
  selectFeatureAreaState,
  (state: PeerMainState) => state.exchangeDashboard
);

export const selectExchangeJobMappingState = createSelector(
  selectFeatureAreaState,
  (state: PeerMainState) => state.exchangeJobMapping
);

export const selectExchangeJobMappingInfoState = createSelector(
  selectFeatureAreaState,
  (state: PeerMainState) => state.exchangeJobMappingInfo
);

// Exchange Request Selectors
export const selectPeerParticipantsState = createSelector(
  selectFeatureAreaState,
  (state: PeerMainState) => state.peerParticipants
);

export const selectAccessExchangeRequestState = createSelector(
  selectFeatureAreaState,
  (state: PeerMainState) => state.accessExchangeRequest
);

export const selectPfCompaniesExchangeRequestState = createSelector(
  selectFeatureAreaState,
  (state: PeerMainState) => state.pfCompaniesExchangeRequest
);

// Exchange Selectors
export const getExchange = createSelector(
  selectExchangeState,
  fromExchangeReducer.getExchange
);
export const getExchangeLoading = createSelector(
  selectExchangeState,
  fromExchangeReducer.getLoading
);
export const getExchangeLoadingError = createSelector(
  selectExchangeState,
  fromExchangeReducer.getLoadingError
);

// Exchange Dashboard Selectors
export const getExchangeDashboardIndustryChartItems = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getIndustryChartItems
);
export const getExchangeDashboardLoadingIndustryChart = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingIndustryChart
);
export const getExchangeDashboardLoadingIndustryChartError = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingIndustryChartError
);

// Exchange Job Mapping Selectors
export const selectExchangeJobMappingsFeatureState = createSelector(
  selectExchangeJobMappingState,
  (state: IFeatureGridState<fromExchangeJobMappingGridReducer.State>) => state.feature
);

export const selectExchangeJobMappingsGridState = createSelector(
  selectExchangeJobMappingState,
  (state: IFeatureGridState<fromExchangeJobMappingGridReducer.State>) => state.grid
);

export const {
  selectAll: getExchangeJobMappings
} = fromExchangeJobMappingGridReducer.adapter.getSelectors(selectExchangeJobMappingsFeatureState);

export const getExchangeJobMappingsLoading = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getLoading
);

export const getExchangeJobMappingsLoadingError = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getLoadingError
);

export const getExchangeJobMappingsTotal = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getTotal
);

export const getExchangeJobMappingsGridState = createSelector(
  selectExchangeJobMappingsGridState,
  fromGridReducer.getGridState
);

export const getExchangeJobMappingsQuery = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getQuery
);

export const getSelectedExchangeJobMapping = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getSelectedMapping
);

export const getExchangeJobMappingPageRowIndexToScrollTo = createSelector(
  selectExchangeJobMappingsFeatureState,
  fromExchangeJobMappingGridReducer.getPageRowIndexToScrollTo
);

export const getExchangeJobMappingsGridData = createSelector(
  getExchangeJobMappings,
  getExchangeJobMappingsTotal,
  (data, total) => {
    return { data: data, total: total };
  }
);

export const getExchangeJobMappingGridStateAndQuery = createSelector(
  getExchangeJobMappingsGridState,
  getExchangeJobMappingsQuery,
  (gridState, query)  => {
    return { gridState, query };
  }
);

// Exchange Job Mapping Info Selectors
export const getCompanyJobsToMapToLoading = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getLoading
);

export const getCompanyJobsToMapToLoadingError = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getLoadingError
);

export const getExchangeJobsInfoApplyingMapping = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getApplyingMapping
);

export const getExchangeJobsInfoApplyingMappingError = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getApplyingMappingError
);

export const getExchangeJobsInfoSelectedMappingCompanyJobId = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getSelectedMappingCompanyJobId
);

export const getExchangeJobsInfoEditingMapping = createSelector(
  selectExchangeJobMappingInfoState,
  fromExchangeJobMappingInfoReducer.getEditingMapping
);

export const {
  selectAll: getCompanyJobsToMapTo
} = fromExchangeJobMappingInfoReducer.adapter.getSelectors(selectExchangeJobMappingInfoState);

// Exchange Request - Access
export const {
  selectAll: getAccessExchangeRequestCandidates
} = fromAccessExchangeRequestReducer.adapter.getSelectors(selectAccessExchangeRequestState);
export const getAccessExchangeRequestCandidatesLoading = createSelector(
  selectAccessExchangeRequestState,
  fromExchangeRequestReducer.getLoading
);
export const getAccessExchangeRequestCandidatesLoadingError = createSelector(
  selectAccessExchangeRequestState,
  fromExchangeRequestReducer.getLoadingError
);
export const getAccessExchangeRequestModalOpen = createSelector(
  selectAccessExchangeRequestState,
  fromExchangeRequestReducer.getModalOpen
);
export const getAccessExchangeRequestRequesting = createSelector(
  selectAccessExchangeRequestState,
  fromExchangeRequestReducer.getRequesting
);
export const getAccessExchangeRequestContext = createSelector(
  selectAccessExchangeRequestState,
  fromAccessExchangeRequestReducer.getLoadingRequestContext
);

// Exchange Request - Access - Peer Participants
export const {
  selectAll: getPeerParticipants
} = fromPeerParticipantsReducer.adapter.getSelectors(selectPeerParticipantsState);

export const getPeerParticipantsLoading = createSelector(
  selectPeerParticipantsState,
  fromPeerParticipantsReducer.getLoading
);

// Exchange Request - pfCompanies
export const {
  selectAll: getPfCompaniesExchangeRequestCandidates
} = fromPfCompaniesExchangeRequestReducer.adapter.getSelectors(selectPfCompaniesExchangeRequestState);
export const getPfCompaniesExchangeRequestCandidatesLoading = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromExchangeRequestReducer.getLoading
);
export const getPfCompaniesExchangeRequestCandidatesLoadingError = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromExchangeRequestReducer.getLoadingError
);
export const getPfCompaniesExchangeRequestModalOpen = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromExchangeRequestReducer.getModalOpen
);
export const getPfCompaniesExchangeRequestRequesting = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromExchangeRequestReducer.getRequesting
);
export const getPfCompaniesExchangeRequestContext = createSelector(
  selectPfCompaniesExchangeRequestState,
  selectExchangeState,
  (exchangeRequestState, exchangeState) => {
    return {
      query: exchangeRequestState.searchTerm,
      exchangeId: exchangeState.exchange ? exchangeState.exchange.ExchangeId : 0
    };
  }
);
