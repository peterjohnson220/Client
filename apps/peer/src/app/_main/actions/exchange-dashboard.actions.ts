import { Action } from '@ngrx/store';

import { GetChartRequest, ChartItem } from 'libs/models';

export const LOADING_CHART  = '[Peer Main/Exchange Dashboard] Loading Chart';
export const LOADING_CHART_SUCCESS  = '[Peer Main/Exchange Dashboard] Loading Chart Success';
export const LOADING_CHART_ERROR  = '[Peer Main/Exchange Dashboard] Loading Chart Error';

export class LoadingChart implements Action {
  readonly type = LOADING_CHART;

  constructor(public payload: GetChartRequest) {}
}

export class LoadingChartSuccess implements Action {
  readonly type = LOADING_CHART_SUCCESS;

  constructor(public payload: ChartItem[]) {}
}

export class LoadingChartError implements Action {
  readonly type = LOADING_CHART_ERROR;
}

export type Actions
  = LoadingChart
  | LoadingChartSuccess
  | LoadingChartError;
