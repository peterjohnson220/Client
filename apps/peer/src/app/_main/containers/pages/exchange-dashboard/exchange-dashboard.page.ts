import { Component, OnInit } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';
import 'hammerjs';

import { Exchange, GetChartRequest, ChartItem } from 'libs/models';

import * as fromExchangeDashboardReducer from '../../../reducers';
import * as fromExchangeDashboardActions from '../../../actions/exchange-dashboard.actions';

@Component({
  selector: 'pf-exchange-dashboard-page',
  templateUrl: './exchange-dashboard.page.html',
  styleUrls: ['./exchange-dashboard.page.scss']
})
export class ExchangeDashboardPageComponent implements OnInit {
  exchangeId: number;
  exchange$: Observable<Exchange>;
  industryChartItems$: Observable<ChartItem[]>;

  constructor(
    private route: ActivatedRoute,
    private router: Router,
    private store: Store<fromExchangeDashboardReducer.State>
  ) {
    this.exchange$ = this.store.select(fromExchangeDashboardReducer.getExchange);
    this.industryChartItems$ = this.store.select(fromExchangeDashboardReducer.getExchangeDashboardIndustryChartItems);
    this.exchangeId = this.route.snapshot.params.id;
  }

  ngOnInit() {
    const getChartRequest: GetChartRequest = {
      ExchangeId: this.exchangeId,
      ChartType: 'Industry'
    };
    this.store.dispatch(new fromExchangeDashboardActions.LoadingChart(getChartRequest));
  }

  manageJobsClick(): void {
    this.router.navigate([ 'exchange/job-mapping', this.exchangeId ]);
  }

  labelContent(e: any): string {
    return e.category;
  }
}
