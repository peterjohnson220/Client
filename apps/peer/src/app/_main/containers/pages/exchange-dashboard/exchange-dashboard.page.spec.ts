import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { Router, ActivatedRoute } from '@angular/router';

import { StoreModule, Store, combineReducers } from '@ngrx/store';
import { of } from 'rxjs/observable/of';
import spyOn = jest.spyOn;

import * as fromRootState from 'libs/state/state';
import {  generateMockExchange } from 'libs/models';

import { ExchangeDashboardPageComponent } from '../exchange-dashboard/exchange-dashboard.page';
import * as fromExchangeDashboardActions from '../../../actions/exchange-dashboard.actions';
import * as fromPeerMainReducer from '../../../reducers';

describe('Peer - Exchange Dashboard', () => {
  let fixture: ComponentFixture<ExchangeDashboardPageComponent>;
  let instance: ExchangeDashboardPageComponent;
  let router: Router;
  let store: Store<fromPeerMainReducer.State>;
  let activatedRoute: ActivatedRoute;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peerMain: combineReducers(fromPeerMainReducer.reducers)
        }),
      ],
      providers: [
        {
          provide: Router,
          useValue: { navigate: jest.fn() },
        },
        {
          provide: ActivatedRoute,
          useValue: { snapshot: { params: { id : 1 } } },
        },
      ],
      declarations: [
        ExchangeDashboardPageComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.get(Store);
    router = TestBed.get(Router);
    activatedRoute = TestBed.get(ActivatedRoute);

    fixture = TestBed.createComponent(ExchangeDashboardPageComponent);
    instance = fixture.componentInstance;

    spyOn(store, 'dispatch');
  });

  it('should show the dashboard with the exchange name and a container for the charts', () => {
    instance.exchange$ = of(generateMockExchange());

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should navigate to the job mapping page when clicking the manage jobs button', () => {
    spyOn(router, 'navigate');

    instance.manageJobsClick();

    expect(router.navigate).toHaveBeenCalledWith(['exchange/job-mapping', activatedRoute.snapshot.params.id]);
  });

  it('should dispatch a industries loadingChart action on init', () => {
    instance.exchange$ = of(generateMockExchange());
    const action = new fromExchangeDashboardActions.LoadingChart({
      ExchangeId: activatedRoute.snapshot.params.id,
      ChartType: 'Industry'
    });

    fixture.detectChanges();

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });
});
