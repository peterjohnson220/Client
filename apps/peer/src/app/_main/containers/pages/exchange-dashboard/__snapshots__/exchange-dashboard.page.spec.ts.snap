// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peer - Exchange Dashboard should show the dashboard with the exchange name and a container for the charts 1`] = `
<pf-exchange-dashboard-page
  exchange$={[Function ScalarObservable]}
  exchangeId={[Function Number]}
  industryChartItems$={[Function Store]}
  route={[Function Object]}
  router={[Function Object]}
  store={[Function Store]}
>
  <pf-page-with-title-bar>
    
  
    
    
    
      Mock Exchange - Dashboard
    
    
  
    
    
    
      
    <button
      class="btn btn-primary"
      type="button"
    >
      Invite Company
    </button>
    
      
    <button
      class="btn btn-primary"
      type="button"
    >
      Manage Jobs
    </button>
    
    
    
  
    
    
    
      
    <div
      class="flex-column-container"
    >
      
        
      <div
        class="card w-100 h-100"
      >
        
          
        <div
          class="card-header"
        >
          
            Industries
          
        </div>
        
          
        <div
          class="card-body"
        >
          
            
          <kendo-chart>
            
              
            <kendo-chart-legend
              position="bottom"
            />
            
              
            <kendo-chart-area
              background="none"
            />
            
              
            <kendo-chart-tooltip>
              
                
              
              
              
            </kendo-chart-tooltip>
            
              
            <kendo-chart-series>
              
                
              <kendo-chart-series-item
                categoryfield="Category"
                field="Value"
                type="donut"
              >
                
                
              </kendo-chart-series-item>
              
              
            </kendo-chart-series>
            
            
          </kendo-chart>
          
          
        </div>
        
        
      </div>
      
      
    </div>
    
    
    
  
  
  </pf-page-with-title-bar>
  
  <pf-payfactors-company-modal />
  
</pf-exchange-dashboard-page>
`;
